src/Approximations/overapproximate_spz.jl

```diff
@@ -0,0 +1,358 @@
+
+### Input
+
+- `lm` -- a linear map of a sparse polynomial zonotope through a matrix zonotope
+
+### Output
+
+A sparse polynomial zonotope overapproximating the linear map.
+
+"""
+function overapproximate(lm::LinearMap{N,S,NM,MAT},
+                         ::Type{<:SparsePolynomialZonotope}) where {N,S<:SparsePolynomialZonotope{N},
+                                                                  NM,
+                                                                  MAT<:MatrixZonotope{NM}}
+    MZ = matrix(lm)
+    P = set(lm)
+    T = promote_type(N, NM)
+
+    m, n = size(MZ)
+    w = ngens(MZ)
+    h = ngens_dep(P)
+    q = ngens_indep(P)
+
+    c = center(MZ) * center(P)
+
+    # matrix of independent generators
+    Gi = Matrix{T}(undef, m, q * (w + 1))
+    Gi[:, 1:q] = center(MZ) * genmat_indep(P)
+
+    # compute matrix of dependent generators
+    G = Matrix{T}(undef, m, h + w + h * w)
+    G[:, 1:h] = center(MZ) * genmat_dep(P)
+
+    # loop to populate G and Gi
+    @inbounds for (i, A) in enumerate(generators(MZ))
+        G[:, h + i] = A * center(P)
+        G[:, (h + w + (i - 1) * h + 1):(h + w + i * h)] = A * genmat_dep(P)
+        Gi[:, (q * i + 1):(q * (i + 1))] = A * genmat_indep(P)
+    end
+
+    # compute exponent
+    Imat = Matrix{Int}(I, w, w)
+    Ê₁, Ê₂, idx = merge_id(indexvector(P), indexvector(MZ), expmat(P), Imat)
+    pₖ = size(Ê₁, 1)
+    E = Matrix{Int}(undef, pₖ, h + w + h * w)
+    E[:, 1:h] = Ê₁
+    E[:, (h + 1):(h + w)] = Ê₂
+    ones_h = ones(Int, 1, h)
+    @inbounds for l in 1:w
+        cstart = (h + w) + (l - 1) * h + 1
+        cend = (h + w) + l * h
+        col = @view Ê₂[:, l]
+        E[:, cstart:cend] = col * ones_h .+ Ê₁
+    end    
+
+    return SparsePolynomialZonotope(c, G, Gi, E, idx)
+end
+
+"""
+	function overapproximate(lm::LinearMap{N,S,NM,MAT},
+                         ::Type{Zonotope}) where {N,S<:AbstractZonotope{N},NM,
+                                                  MAT<:MatrixZonotope{NM}}
+
+Overapproximate the linear map of a zonotope through a matrix zonotope,
+following a modification of Proposition 1 of [HuangLBS2025](@citet).
+
+### Input
+
+- `lm` -- a linear map of a zonotope through a matrix zonotope
+
+### Output
+
+A zonotope overapproximating the linear map.
+
+"""
+function overapproximate(lm::LinearMap{N,S,NM,MAT},
+                         ::Type{<:Zonotope}) where {N,S<:AbstractZonotope{N},NM,
+                                                  MAT<:MatrixZonotope{NM}}
+    MZ = matrix(lm)
+    Z = set(lm)
+    T = promote_type(N, NM)
+
+    m, n = size(MZ)
+    w = ngens(MZ)
+    h = ngens(Z)
+
+    c = center(MZ) * center(Z)
+
+    # compute matrix of dependent generators
+    G = Matrix{T}(undef, m, h + w + h * w)
+    G[:, 1:h] = center(MZ) * genmat(Z)
+    @inbounds for (i, A) in enumerate(generators(MZ))
+        G[:, h + i] = A * center(Z)
+        G[:, (h + w + (i - 1) * h + 1):(h + w + i * h)] = A * genmat(Z)
+    end
```

Comment:
I assumed that you want to imitate the code for the `GI` matrix of independent generators. Why the dependent ones?

Reply:
It is not about the name but about what the code is doing. I think it is different.